### PR TITLE
Add automatic travis deployment step

### LINKED ISSUE #32: Improve versioning process
Today, versioning is done manually by changing the `__version__.py` file and then running the `make release` command.

Versioning should be a part of the release process.

### PR DESCRIPTION
**Pull request by destag**
After pushing tag to git this will deploy new version of package to pypi.

Closes #32

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,4 +11,18 @@ install:
   - pip install -r requirements_dev.txt
 # command to run tests
 script:
-  - make test
+- make test
```
Comment (dmvieira) [inline .travis.yml]: Please indent this one

File: .travis.yml [hunk 2]
```diff
@@ -11,4 +11,18 @@ install:
...
+
+before_deploy: sed -i "s/__version__ = \".*\"/__version__ = \"$TRAVIS_TAG\"/g" azkaban_cli/__version__.py
+
+deploy:
+  provider: pypi
+  user: $PYPI_USER
+  password: $PYPI_PASSWORD
```
Comment (dmvieira) [inline .travis.yml:21]: Thank you for this great work!

We should use encrypted password as environment variable
```
deploy:
  provider: pypi
  user: "Your username"
  password:
    secure: "Your encrypted password"
```

**Review by dmvieira — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (destag) [reply .travis.yml]: Sorry, my mistake when rebasing

File: .travis.yml (hunk 2 above)
Comment (destag) [reply .travis.yml:21]: Done, you will have to encrypt password using [travis encrypt](https://docs.travis-ci.com/user/encryption-keys#usage)

File: .travis.yml [hunk 3]
```diff
@@ -12,3 +12,18 @@ install:
 # command to run tests
 script:
   - make test
+
+before_deploy: sed -i "s/__version__ = \".*\"/__version__ = \"$TRAVIS_TAG\"/g" azkaban_cli/__version__.py
+
+deploy:
+  provider: pypi
+  user: "Your username"
```
Comment (dmvieira) [inline .travis.yml]: Please come back with environment variable... Can I use environment variables that are in Travis here?

File: .travis.yml [hunk 4]
```diff
@@ -12,3 +12,18 @@ install:
...
+  password:
+    secure: "Your encrypted password"
```
Comment (dmvieira) [inline .travis.yml]: If I really can use Travis environment variables here, then come back with password variable. I will encrypt it using Travis and add as an environment variable there

**Review by dmvieira — CHANGES_REQUESTED** [review]
Thank you so much, I think it almost right!

File: .travis.yml (hunk 3 above)
Comment (destag) [reply .travis.yml]: You can set secrets and other environment variables in travis site.

Output of my test pipeline looks like this:

```bash
$ echo $PYPI_PASSWORD
[secure]
```

**Review by dmvieira — APPROVED** [review]

**Pull request merged**